src/chains-config/metadata-consts/index.ts

```diff
@@ -0,0 +1,45 @@
+import { Definition } from '../../types/chains-config';
+import { kusamaDefinitions } from './kusama-consts';
+import { polkadotDefinitions } from './polkadot-consts';
+
+/**
+ * Creates an object that orders each runtime to their appropriate data
+ *
+ * @param definitions An array of objects that group data based on runtimes
+ * and their extrinsicBaseWeight metadata
+ */
+const generateBlockWeightObject = (
+	definitions: Definition[]
+): Record<string, Definition> => {
+	const blockWeightObject = {};
+
+	for (let i = 0; i < definitions.length; i++) {
+		const runtimes: number[] = definitions[i].runtimes;
+
+		for (let p = 0; p < runtimes.length; p++) {
```

Comment:
These double for-loops give me pause. Can you give me an intuition for how big these collections are we're looping over? And why we're convinced they will never ever be too large for this to matter?

Reply:
Based on the current code, the number of iterations == number of known runtimes.

Reply:
Yea of course, double for loops are always a sight for caution. In this case I think it's justifiable, because `i == 2` and wont get any larger. It's is equal to the amount of chains that are hardcoded ie `polkadot`, and `kusama`. Then in the case of `p`, it will only ever be as big as the number of runtime's that exists for Polkadot or Kusama. So that number will only grow by probabaly no more than 12 a year.

Reply:
If `i` is known to be 2 maybe we can just hardcode the two? Maybe it's easier to read that way?

Reply:
So I actually completely re worked this function, I still use a double for loop, but I think you'll find it to be a lot more readable